disnake/ext/commands/params.py

```diff
@@ -787,7 +756,12 @@ def parse_annotation(self, annotation: Any, converter_mode: bool = False) -> boo
         return True
 
     def parse_converter_annotation(self, converter: Callable, fallback_annotation: Any) -> None:
-        _, parameters = isolate_self(signature(converter))
+        if isinstance(converter, (types.FunctionType, types.MethodType)):
+            converter_func = converter
+        else:
+            # if converter isn't a function/method, it should be a callable object/type
+            converter_func = converter.__call__
```

Comment:
Is this done to ensure that the `converter` is callable? I'm asking this because `converter(...)` and `converter.__call__(...)` give the same result

Reply:
Yeah, kind of - `converter.__call__` is only used for signatures, running the converter still just calls `converter()`.
That snippet originated from [here](https://github.com/DisnakeDev/disnake/pull/1116/files#diff-6377fad5f75948c1eaa880c932839dd3f47b01c21056433ada31632cb518a665L146-L149), but I'm quite certain it's only relevant for converters.

Essentially, we want to pass `converter.__call__` through the annotation code (instead of `converter`), since we need the converter's global namespace for eval'ing its annotations. `inspect.signature` etc. sort of handles this, but we're evaluating annotations ourselves as well.
For functions/methods, this is just `func.__globals__`; for classes, `inspect` uses `sys.modules[obj.__module__].__dict__`, we can get the same dict through `obj.__call__.__globals__`.

Reply:
Tried to clarify this a bit in bb03995289322956ce936af6f154d90ddad58ac0, I hope it makes more sense now :p